listeners_interface_test.go

```diff
@@ -0,0 +1,507 @@
+// Copyright 2025 Matthew Holt and The Caddy Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caddy
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestIsInterfaceName(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected bool
+		desc     string
+	}{
+		// Valid interface names
+		{"eth0", true, "typical ethernet interface"},
+		{"wlan0", true, "wireless interface"},
+		{"tailscale0", true, "tailscale interface"},
+		{"enp0s3", true, "predictable network interface name"},
+		{"lo", true, "loopback interface"},
+		{"docker0", true, "docker bridge interface"},
+
+		// Invalid interface names (IP addresses)
+		{"192.168.1.1", false, "IPv4 address"},
+		{"127.0.0.1", false, "localhost IPv4"},
+		{"::1", false, "IPv6 localhost"},
+		{"2001:db8::1", false, "IPv6 address"},
+
+		// Invalid interface names (hostnames)
+		{"example.com", false, "hostname with dots"},
+		{"localhost", false, "hostname"},
+		{"my-host.local", false, "hostname with dashes and dots"},
+
+		// Invalid interface names (file descriptors)
+		{"3", false, "numeric file descriptor"},
+		{"10", false, "another numeric file descriptor"},
+
+		// Invalid interface names (other)
+		{"", false, "empty string"},
+		{"verylonginterfacenamethatdefinitelyexceedsfiftychars", false, "too long interface name"},
```

Comment:
From some quick googling, it seems to me like interface names are never longer than 15 chars